tests/test_deserialization.py

```diff
@@ -542,6 +542,36 @@ def test_dict_field_deserialization(self):
             field.deserialize('baddict')
         assert excinfo.value.args[0] == 'Not a valid mapping type.'
 
+    def test_structured_dict_value_deserialization(self):
+        field = fields.Dict(values=fields.Str)
+        assert field.deserialize({"foo": "bar"}) == {"foo": "bar"}
+        with pytest.raises(ValidationError) as excinfo:
+            field.deserialize({"foo": 1})
+        assert excinfo.value.args[0] == {'foo': ['Invalid value: Not a valid string.']}
+
+    def test_structured_dict_key_deserialization(self):
+        field = fields.Dict(keys=fields.Str)
+        assert field.deserialize({"foo": "bar"}) == {"foo": "bar"}
+        with pytest.raises(ValidationError) as excinfo:
+            field.deserialize({1: "bar"})
+        assert excinfo.value.args[0] == {1: ['Invalid key: Not a valid string.']}
+
+    def test_structured_dict_key_value_deserialization(self):
+        field = fields.Dict(keys=fields.Str, values=fields.Decimal)
+        assert field.deserialize({"foo": 1}) == {"foo": decimal.Decimal('1')}
+        with pytest.raises(ValidationError) as excinfo:
+            field.deserialize({1: "bar"})
+        assert excinfo.value.args[0] == {1: [
+            'Invalid key: Not a valid string.',
```

Comment:
What does the error data look like if a key has more than one error? It would be worth adding a test that uses a field with a validator. Something like

```python
fields.Dict(keys=fields.Str(validate=[validate.Length(min=2)]))
```